rskj-core/src/test/java/co/rsk/peg/federation/PendingFederationTest.java

```diff
@@ -137,96 +137,153 @@ void testToString() {
     }
 
-    @Test
-    void buildFederation_after_RSKIP_353_activation_testnet() {
-        testBuildFederation(
-            true,
-            true,
-            true,
-            NetworkParameters.ID_TESTNET,
-            6
+    @ParameterizedTest
+    @MethodSource("networkParameters")
+    void buildFederation_with6Members_afterRSKIP353Activation_BeforeRSKIP305Activations_shouldBuildP2SHERPFed(NetworkParameters networkParams, FederationConstants federationConstants) {
+        // Arrange
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.hop401().forBlock(FEDERATION_CREATION_BLOCK_NUMBER);
+        FederationArgs federationArgs = new FederationArgs(federationMembers, creationTime, FEDERATION_CREATION_BLOCK_NUMBER, networkParams);
+        List<BtcECKey> erpPubKeys = federationConstants.getErpFedPubKeysList();
+        long activationDelay = federationConstants.getErpFedActivationDelay();
+        Federation expectedFederation = FederationFactory.buildP2shErpFederation(federationArgs, erpPubKeys, activationDelay);
+
+        // Act
+        PendingFederation otherPendingFederation = new PendingFederation(
+            federationMembers
         );
+        Federation builtFederation = otherPendingFederation.buildFederation(
+            creationTime,
+            FEDERATION_CREATION_BLOCK_NUMBER,
+            federationConstants,
+            activations
+        );
+
+        // Assert
+        assertEquals(expectedFederation, builtFederation);
     }
 
-    @Test
-    void buildFederation_after_RSKIP_353_activation_mainnet() {
-        testBuildFederation(
-            true,
-            true,
-            true,
-            NetworkParameters.ID_MAINNET,
-            6
+    @ParameterizedTest
+    @MethodSource("networkParameters")
+    void buildFederation_with6Members_afterRSKIP305Activation_shouldBuildP2SHP2WSHERPFed(NetworkParameters networkParams, FederationConstants federationConstants) {
+        // Arrange
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.tbd800().forBlock(FEDERATION_CREATION_BLOCK_NUMBER);
+        FederationArgs federationArgs = new FederationArgs(federationMembers, creationTime, FEDERATION_CREATION_BLOCK_NUMBER, networkParams);
+        List<BtcECKey> erpPubKeys = federationConstants.getErpFedPubKeysList();
+        long activationDelay = federationConstants.getErpFedActivationDelay();
+        Federation expectedFederation = FederationFactory.buildP2shP2wshErpFederation(federationArgs, erpPubKeys, activationDelay);
+
+        // Act
+        PendingFederation otherPendingFederation = new PendingFederation(
+            federationMembers
         );
+        Federation builtFederation = otherPendingFederation.buildFederation(
+            creationTime,
+            FEDERATION_CREATION_BLOCK_NUMBER,
+            federationConstants,
+            activations
+        );
+
+        // Assert
+        assertEquals(expectedFederation, builtFederation);
```

Comment:
I think we should add one more assertion, at least checking that the resulting federation has the correct type. Wdyt?

Reply:
I think it's not worth it because we are using the method `FederationFactory.buildP2shP2wshErpFederation(federationArgs, erpPubKeys, activationDelay)` as the expectedFed. 

The test to check that it returns the correct type is tested by https://github.com/rsksmart/rskj/blob/280c7a3a9d73398b69c7793d6566e0cda5ba901e/rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java#L191